zha-network-card.js

```diff
@@ -310,22 +310,54 @@ class ZHANetworkCard extends HTMLElement {
     });
   }
 
-  set hass(hass) {
-    const config = this._config;
-    const root = this.shadowRoot;
-
-    hass
+  async init(hass, config, root) {
+    await hass
       .callWS({
         type: "zha/devices",
       })
-      .then((devices) => {
+      .then(async (devices) => {
         // `raw_rows` to be filled with data here, due to 'attr_as_list' it is possible to have
         // multiple data `raw_rows` acquired into one cell(.raw_data), so re-iterate all rows
         // to---if applicable---spawn new DataRowZHA objects for these accordingly
-        let raw_rows = devices.map(
+        const raw_rows = devices.map(
           (e) => new DataRowZHA({ attributes: e }, config.strict)
         );
-        raw_rows.forEach((e) => e.get_raw_data(config.columns));
+        const read_sw_build_id = config.columns.filter((col) => col.prop == "sw_build_id");
+        await Promise.allSettled(raw_rows.map(async (e) => {
+          if (read_sw_build_id && e.device.attributes.available) {
+            // retrieving cluster attributes requires additional ws calls
+            await hass.callWS({
```

Comment:
These `hass.callWS()` invocations of `zha/devices/clusters` and `zha/devices/clusters/attributes/value` types could probably be cached to avoid hammering the zigbee network and devices.